polkadot-cli/src/lib.rs

```diff
@@ -53,8 +63,29 @@ pub fn run<I, T>(args: I) -> error::Result<()> where
 	init_logger(log_pattern);
 
 	// Create client
-	let executor = executor::executor();
-	let client = client::new_in_mem(executor)?;
+	let executor = polkadot_executor::executor();
+	let mut storage = Default::default();
+	let god_key = hex!["3d866ec8a9190c8343c2fc593d21d8a6d0c5c4763aaab2349de3a6111d64d124"];
+
+	use polkadot_runtime::genesismap::{additional_storage_with_genesis, GenesisConfig};
+	let genesis_config = GenesisConfig {
+		validators: vec![god_key.clone()],
+		authorities: vec![god_key.clone()],
+		balances: vec![(god_key.clone(), 1u64 << 63)].into_iter().collect(),
+		block_time: 5,			// 5 second block time.
+		session_length: 720,	// that's 1 hour per session.
+		sessions_per_era: 24,	// 24 hours per era.
+		bonding_duration: 90,	// 90 days per bond.
+		approval_ratio: 667,	// 66.7% approvals required for legislation.
+	};
+	let prepare_genesis = || {
+		storage = genesis_config.genesis_map();
+		let block = genesis::construct_genesis_block(&storage);
+		storage.extend(additional_storage_with_genesis(&block).into_iter());
+		use codec::Slicable;
+		(primitives::block::Header::from_slice(&mut block.header.to_vec().as_ref()).expect("to_vec() always gives a valid serialisation; qed"), storage.into_iter().collect())
```

Comment:
there has got to be a better way to convert in-memory than serializing and deserializing

Reply:
not really. right now, they're equivalent, but eventually (e.g. when we introduce `parachain_activity`) they'll only be compatible in terms of binary format. we might eventually do a specialised `From` implementation, but it's hardly critical path.